components/page-templates/B2B/QuoteDetailsTemplate/QuoteDetailsTemplate.tsx

```diff
@@ -504,6 +509,31 @@ const QuoteDetailsTemplate = (props: QuoteDetailsTemplateProps) => {
       console.error(e)
     }
   }
+  const handleApplyCouponCode = async (couponCode: string) => {
+    try {
+      setPromoError('')
+      const response = await updateQuoteCoupon.mutateAsync({
+        quoteId,
+        couponCode,
+      })
+      if (response?.invalidCoupons?.length) {
+        setPromoError(response?.invalidCoupons[0]?.reason)
+      }
+    } catch (err) {
+      console.error(err)
```

Comment:
instead of console log we should show error message coming from api using toster.